File: Wikipedia/Categories/UIViewController+WMFHideKeyboard.m

#import "UIViewController+WMFHideKeyboard.h"

@implementation UIViewController (WMFHideKeyboard)

- (void)wmf_hideKeyboard {
    //http://stackoverflow.com/questions/11879745/an-utility-method-for-hiding-the-keyboard
    [[UIApplication sharedApplication] sendAction:@selector(resignFirstResponder) to:nil from:nil forEvent:nil];
}

@end

File: Wikipedia/Code/UIViewController+WMFHideKeyboard.swift
extension UIViewController {
    /**
     *  Uses the responder chain to make all UIResponders
     *  in the view hierarchy resignFirstResponder.
     */
    @objc func wmf_hideKeyboard() {
        UIApplication.shared.sendAction(#selector(UIApplication.resignFirstResponder), to: nil, from: nil, for: nil)
    }
}
